src/AuthenticationService.php

```diff
@@ -255,4 +256,34 @@ public function getResult()
     {
         return $this->_result;
     }
+
+    /**
+     * Gets an identity object
+     *
+     * @return null|\Authentication\IdentityInterface
+     */
+    public function getIdentity()
+    {
+        if (!$this->_result->isValid()) {
+            return null;
+        }
+
+        return $this->buildIdentity($this->_result->getIdentity());
+    }
+
+    /**
+     * Builds the identity object
+     *
+     * @param array|ArrayAccess $identityData Identity data
+     * @return \Authentication\IdentityInterface
+     */
+    public function buildIdentity($identityData)
+    {
+        $class = $this->getConfig('identityClass');
+        if (is_callable($class)) {
+            return $class($identityData);
+        }
+
+        return new $class($identityData);
```

Comment:
This is a nice approach. Accepting either a class string or a callback factory function is nice :+1: